blockstack/blockstackd.py

```diff
@@ -3482,6 +3499,82 @@ def run_blockstackd():
            print "Failed to sign snapshot"
            sys.exit(1)
 
+    elif args.action == 'export_migration_json':
+        path = str(args.path)
+        rc = fast_sync_import(working_dir, path, public_keys=None, num_required=0, verbose=True, delete_file=False)
+        if not rc:
+           print 'fast_sync failed'
+           sys.exit(1)
+
+        # treat this as a recovery
+        setup_recovery(working_dir)
+
+        # allow consensus_hash to be provided as a string or within a file
+        consensus_hash = args.consensus_hash
+        if os.path.exists(consensus_hash):
+            with open(consensus_hash, 'r') as file:
+                consensus_hash = file.read().replace('\n', '')
+    
+        db = get_db_state(working_dir)
+        ch = db.get_current_consensus()
+        if str(ch) != consensus_hash:
+            print "Fast-sync import does not match consensus hash!"
+            sys.exit(1)
+        block = db.get_current_block()
+
+        tmpdir = tempfile.mkdtemp('blockstack-verify-chainstate-XXXXXX')
+        db_verified = False
+        try:
+            db_verified = verify_database(consensus_hash, block, working_dir, tmpdir)
+        except:
+            pass
+        if db_verified:
+           print "Database is consistent with %s" % args.consensus_hash
+        else:
+           print "Database is NOT CONSISTENT"
+           # TODO: can't get this passing using regtest fastdumps
```

Comment:
Do you have a log snippet?  You should see from the log that the `verify_database()` method re-generates the same sequence of consensus hashes that got produced when the database itself was produced.  You would also see the history of consensus hashes diverge if something was wrong with the snapshot.

Reply:
We will now be testing this on mainnet data on a secondary test node in PBC infra.